File: sach/de3301.cpp
//DE3301.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
using namespace std;
int n,a[10001];
int main()
{
  cin>>n;
  for(int i=1;i<=n;i++)
      cin>>a[i];
  sort(a+1,a+n+1);
  int i=1, T=0;
  while(i<n)
  {
      T++;
      a[i]--;
      if(a[i]==0) i++;
      n--;
  }
  cout<<T;
  return 0;
}

File: sach/de3301.py
//DE3301.PY - Tran Huu Nam - hsgtin.vn - 2024
n=int(input())
A=list(map(int,input().split()))
A.sort()
l,cnt=0,0
while l<n-1:
    A[l]-=1
    if A[l]==0:
        l+=1
    n-=1
    cnt+=1
print(cnt)

